### ISSUE TITLE
Documentation contradictory on namespaceSelector

### ISSUE THREAD

**Issue by Drvanon** [prometheus-operator/prometheus-operator#3613]
**What happened?**
I was trying to understand how I can add a new serviceMonitor to prometheus.

**Did you expect to see something different?**
I found the statement 'to discover targets in all namespaces the namespaceSelector has to be empty' to conflict with the piece of code that follows. Should namespaceSelector now be empty, or should it be an object with the code that follows after it?

**Environment**
Github on latest commit was 10206b0 on the file Documentation/design.md

**Comment by coderanger** [comment]
Just so everyone doesn't have to go find it, the following code is:

```
spec:
  namespaceSelector:
    any: true
```

Which is indeed incorrect. Use `{}` as the value, i.e. an empty object. I think this is just from an old version and never got updated, thanks!

**Comment by nrchakradhar** [comment]
The API docs also mention that `any should be set to true`
https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#namespaceselector.
I think if not design, API docs should be atleast updated.

**Comment by paulfantom** [comment]
Sadly prometheus-operator API is not consistent on namespace selectors and we have [a long-standing TODO item to fix this](https://github.com/prometheus-operator/prometheus-operator/blob/master/pkg/apis/monitoring/v1/types.go#L1331-L1333). Right now prometheus-operator is using almost everywhere usual Kubernetes label selectors as namespace selectors and in case of those setting value to `{}` allows to discovery everything. However `ServiceMonitorSpec.namespaceSelector`, `PodMonitorSpec.namespaceSelector`, and `ProbeTargetIngress.namespaceSelector` are not using Kubernetes label selectors and setting `any: true` to select all namespaces.

In short if selector is named exactly as `namespaceSelector` then it needs `any: true` and if it is named as `**STH**NamespaceSelector` (where `**STH**` is for example `rule`) then it needs `{}`.

Note: api.md doc file is autogenerated from code, so there is not much possibility for it to go stale.

**Comment by paulfantom** [comment]
Possibly duplicate of #3553
